activerecord/lib/active_record/transactions.rb

```diff
@@ -206,6 +206,19 @@ def transaction(options = {}, &block)
         connection.transaction(options, &block)
       end
 
+      def start_sandbox
+        @sandbox_connections = connection_handler.connection_pools
+          .map(&:connection)
+          .each {|c| c.begin_transaction joinable: false }
+      end
+
+      def finish_sandbox
+        if instance_variable_defined?(:@sandbox_connections) && @sandbox_connections
```

Comment:
Minor - this will result starting and stopping sandbox to leave with `sandbox_connections` instance var, but not before it's started for first time.

For consistency, perhaps ether define `@sandbox_connections = nil` at class init, or use `remove_instance_variable(@sandbox_connections)`